app.py

```diff
@@ -38,438 +33,452 @@
 
 logging.getLogger('PIL').setLevel(logging.WARNING)
 
-FILTER_DICT = {'Ha': 0, 'B': 1, 'V': 2, 'g': 3, 'r': 4, 'i': 5}
-FILTER_DICT_REVERSE = {0: 'Ha', 1: 'B', 2: 'V', 3: 'g', 4: 'r', 5: 'i'}
-
-DEFAULT_PATH = '/data/ecam'
-
-# If we're debugging, use a local directory instead - create if doesn't exist
-if DEBUGGING:
-    DEFAULT_PATH = './' + DEFAULT_PATH
-    os.makedirs(os.path.dirname(DEFAULT_PATH), exist_ok=True)
-
-DUMMY_FILTER_POSITION = 0
-
-ABORT_FLAG = False
-
-def getFilePath(file):
-    """
-    Formats the given file name to be valid.
-    If the file contains invalid characters or is empty, image.fits will be used.
-    if the file already exists, it will be saved as:
-        name(0), name(1), name(2), ..., name(n)
-    """
-
-    default_image_name = "ecam-{seq:04d}.fits"
-
-    date = Time.now().utc.isot.split("T")[0].replace("-", "")
-
-    path = os.path.join(DEFAULT_PATH, date)
-    os.makedirs(path, exist_ok=True)
-
-    invalid_characters = [":", "<", ">", "/", "\\", '"', "|", "?", "*", ".."]
-    # if invalid filename, use image.fits
-    if file is None or file == "" or any(c in file for c in invalid_characters):
-        all_files = list(sorted(glob(os.path.join(path, "ecam-*.fits"))))
-        if len(all_files) == 0:
-            seq = 1
-        else:
-            match = re.search(r"ecam\-([0-9]+)", all_files[-1])
-            if match:
-                seq = int(match.group(1)) + 1
-            else:
+class EvoraServer:
+    """EvoraServer class containing functions to interface with the Evora CCD."""
+
+    def __init__(self, args):
+        """This function will run once when EvoraServer is created."""
+        # Interpret arguments as globals
+        self.DEBUGGING = args.debug
+        self.PORT = args.port
+        self.DEFAULT_PATH = args.path
+
+        # Constants + globals
+        self.ABORT_FLAG = False
+        self.DUMMY_FILTER_POSITION = 0
+        self.FILTER_DICT = {'Ha': 0, 'B': 1, 'V': 2, 'g': 3, 'r': 4, 'i': 5}
+        self.FILTER_DICT_REVERSE = {0: 'Ha', 1: 'B', 2: 'V', 3: 'g', 4: 'r', 5: 'i'}
+
+        # If we're debugging, use a local directory instead - create if doesn't exist
```

Comment:
I know this wasn't a change you made, but we should have this documented that using debug mode will save in a local directory. I wonder if we even need this functionality?